File: PST/FUEL Source Code/FUEL/modCheckForUpdates.cs
// Decompiled with JetBrains decompiler
// Type: FUEL.modCheckForUpdates
// Assembly: FUEL, Version=2.4.8.0, Culture=neutral, PublicKeyToken=null
// MVID: A3D630D0-0620-425D-8CFB-B67B2AC75507
// Assembly location: C:\hp\FUEL\FUEL.dll

using FUEL.My;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using RDL;
using System;
using System.IO;
using System.Threading;
using System.Windows.Forms;

namespace FUEL
{
  [StandardModule]
  internal sealed class modCheckForUpdates
  {
    private static Thread thdUpdateCheck;

    internal static void CheckForUpdate(FUEL.CheckForUpdates.CheckType FUELType)
    {
      try
      {
        bool flag = true;
        try
        {
          if (modCheckForUpdates.thdUpdateCheck.IsAlive)
            flag = false;
        }
        catch (NullReferenceException ex)
        {
          ProjectData.SetProjectError((Exception) ex);
          flag = true;
          ProjectData.ClearProjectError();
        }
        if (!new NetworkConnection().CheckForNetworkConnection((RichTextBox) null))
          flag = false;
        if (!flag || DateAndTime.DateDiff("h", (object) MySettingsProperty.Settings.UpdateCheck.ToString(), (object) DateAndTime.Now) <= 24L)
          return;
        modCheckForUpdates.thdUpdateCheck = new Thread((ParameterizedThreadStart) (a0 => modCheckForUpdates.CheckForUpdates((FUEL.CheckForUpdates.CheckType) Conversions.ToInteger(a0))));
        modCheckForUpdates.thdUpdateCheck.IsBackground = true;
        modCheckForUpdates.thdUpdateCheck.Start((object) FUELType);
      }
      catch (Exception ex)
      {
        ProjectData.SetProjectError(ex);
        int num = (int) Interaction.MsgBox((object) ("Error while checking for update.\r\n\r\n" + ex.ToString()));
        ProjectData.ClearProjectError();
      }
    }

    private static void CheckForUpdates(FUEL.CheckForUpdates.CheckType FUELType)
    {
      MySettingsProperty.Settings.UpdateCheck = DateAndTime.Now;
      MySettingsProperty.Settings.Save();
      string str = "\\\\vcslab.vcd.hp.com\\root\\InkSystems\\SPHINKS\\Randal Morrison\\FUEL\\VersionInfo.txt";
      string SaveAs = Path.Combine(modCommonCode.GetDataPath(), "VersionInfo.txt");
      FUEL.CheckForUpdates checkForUpdates = new FUEL.CheckForUpdates(str, SaveAs, FUELType);
      if (!Operators.ConditionalCompareObjectNotEqual(checkForUpdates.UpdateAvailable, (object) FUEL.CheckForUpdates.UpdateType.None, false))
        return;
      string Prompt = (string) null;
      string Title = "Update Available";
      bool flag = false;
      if (Conversions.ToBoolean(Operators.AndObject(Operators.CompareObjectEqual(checkForUpdates.UpdateAvailable, (object) FUEL.CheckForUpdates.UpdateType.OptionalUpdate, false), (object) (FUELType != FUEL.CheckForUpdates.CheckType.PST))))
      {
        Prompt = "There is a new version of FUEL available.\r\n\r\nInstalled Version: " + checkForUpdates.CurrentVersion + "\r\nNew Version: " + checkForUpdates.ServerVersion + "\r\n\r\nGo to the location below to install the new version:\r\n" + Path.GetDirectoryName(str);
        flag = true;
      }
      else if (Operators.ConditionalCompareObjectEqual(checkForUpdates.UpdateAvailable, (object) FUEL.CheckForUpdates.UpdateType.RequiredUpdate, false))
      {
        Prompt = "There is a CRITICAL new version of FUEL available.\r\n\r\nInstalled Version: " + checkForUpdates.CurrentVersion + "\r\nNew Version: " + checkForUpdates.ServerVersion + "\r\n\r\nCritical updates are reserved for occasions where a significant bug was fixed, or a required new feature was added. Please update your version of FUEL as soon as possible.\r\n\r\nGo to the location below to install the new version:\r\n" + Path.GetDirectoryName(str);
        Title = "Critical Update Available";
        flag = true;
      }
      if (!flag)
        return;
      int num = (int) Interaction.MsgBox((object) Prompt, Title: (object) Title);
    }
  }
}

File: PST/FUEL/src/FUEL/modCheckForUpdates.vb
Imports FUEL.My
Imports Microsoft.VisualBasic
Imports Microsoft.VisualBasic.CompilerServices
Imports RDL
Imports System
Imports System.Diagnostics
Imports System.IO
Imports System.Runtime.CompilerServices
Imports System.Threading

Namespace FUEL
    <StandardModule> _
    Friend NotInheritable Class modCheckForUpdates
        ' Methods
        <DebuggerStepThrough, CompilerGenerated> _
        Private Shared Sub _Lambda$__1(ByVal a0 As Object)
            modCheckForUpdates.CheckForUpdates(DirectCast(Conversions.ToInteger(a0), CheckType))
        End Sub

        Friend Shared Sub CheckForUpdate(ByVal FUELType As CheckType)
            Try 
                Dim flag As Boolean = True
                Try 
                    If modCheckForUpdates.thdUpdateCheck.IsAlive Then
                        flag = False
                    End If
                Catch exception1 As NullReferenceException
                    Dim ex As NullReferenceException = exception1
                    ProjectData.SetProjectError(ex)
                    Dim exception As NullReferenceException = ex
                    flag = True
                    ProjectData.ClearProjectError
                End Try
                If Not New NetworkConnection().CheckForNetworkConnection(Nothing) Then
                    flag = False
                End If
                If (flag AndAlso (DateAndTime.DateDiff("h", MySettingsProperty.Settings.UpdateCheck.ToString, DateAndTime.Now, FirstDayOfWeek.Sunday, FirstWeekOfYear.Jan1) > &H18)) Then
                    modCheckForUpdates.thdUpdateCheck = New Thread(New ParameterizedThreadStart(AddressOf modCheckForUpdates._Lambda$__1))
                    modCheckForUpdates.thdUpdateCheck.IsBackground = True
                    modCheckForUpdates.thdUpdateCheck.Start(FUELType)
                End If
            Catch exception3 As Exception
                Dim ex As Exception = exception3
                ProjectData.SetProjectError(ex)
                Interaction.MsgBox(("Error while checking for update." & ChrW(13) & ChrW(10) & ChrW(13) & ChrW(10) & ex.ToString), MsgBoxStyle.ApplicationModal, Nothing)
                ProjectData.ClearProjectError
            End Try
        End Sub

        Private Shared Sub CheckForUpdates(ByVal FUELType As CheckType)
            MySettingsProperty.Settings.UpdateCheck = DateAndTime.Now
            MySettingsProperty.Settings.Save
            Dim dLLocation As String = "\\vcslab.vcd.hp.com\root\InkSystems\SPHINKS\Randal Morrison\FUEL\VersionInfo.txt"
            Dim updates As New CheckForUpdates(dLLocation, Path.Combine(modCommonCode.GetDataPath, "VersionInfo.txt"), FUELType)
            If Operators.ConditionalCompareObjectNotEqual(updates.UpdateAvailable, UpdateType.None, False) Then
                Dim strArray As String()
                Dim prompt As String = Nothing
                Dim title As String = "Update Available"
                Dim flag As Boolean = False
                If Conversions.ToBoolean(Operators.AndObject(Operators.CompareObjectEqual(updates.UpdateAvailable, UpdateType.OptionalUpdate, False), (FUELType <> CheckType.PST))) Then
                    strArray = New String() { "There is a new version of FUEL available." & ChrW(13) & ChrW(10) & ChrW(13) & ChrW(10) & "Installed Version: ", updates.CurrentVersion, ChrW(13) & ChrW(10) & "New Version: ", updates.ServerVersion, ChrW(13) & ChrW(10) & ChrW(13) & ChrW(10) & "Go to the location below to install the new version:" & ChrW(13) & ChrW(10), Path.GetDirectoryName(dLLocation) }
                    prompt = String.Concat(strArray)
                    flag = True
                ElseIf Operators.ConditionalCompareObjectEqual(updates.UpdateAvailable, UpdateType.RequiredUpdate, False) Then
                    strArray = New String() { "There is a CRITICAL new version of FUEL available." & ChrW(13) & ChrW(10) & ChrW(13) & ChrW(10) & "Installed Version: ", updates.CurrentVersion, ChrW(13) & ChrW(10) & "New Version: ", updates.ServerVersion, ChrW(13) & ChrW(10) & ChrW(13) & ChrW(10) & "Critical updates are reserved for occasions where a significant bug was fixed, or a required new feature was added. Please update your version of FUEL as soon as possible." & ChrW(13) & ChrW(10) & ChrW(13) & ChrW(10) & "Go to the location below to install the new version:" & ChrW(13) & ChrW(10), Path.GetDirectoryName(dLLocation) }
                    prompt = String.Concat(strArray)
                    title = "Critical Update Available"
                    flag = True
                End If
                If flag Then
                    Interaction.MsgBox(prompt, MsgBoxStyle.ApplicationModal, title)
                End If
            End If
        End Sub


        ' Fields
        Private Shared thdUpdateCheck As Thread
    End Class
End Namespace

